File: ros/simple-publisher-Float32MultiArray.cpp
// simple sample for an c++ Float32MultiArray publisher

#include "ros/ros.h"
#include "std_msgs/Float32MultiArray.h"
int main(int argc, char **argv)
{
    ros::init(argc, argv, "Array_pub");
    ros::NodeHandle nh;
 
    ros::Publisher chatter_pub = nh.advertise<std_msgs::Float32MultiArray>("chatter", 1000);
 
    ros::Rate loop_rate(10);
    while (ros::ok())
    {
        std_msgs::Float32MultiArray msg;
        msg.data.push_back(1.0);//自己写的，可行
        msg.data.push_back(2.0);
        msg.data.push_back(3.0);
        msg.data.push_back(4.0);
 
        chatter_pub.publish(msg);
        ros::spinOnce();
        loop_rate.sleep();
    }
    return 0;
}

File: ros/simple-publisher-Float32MultiArray.py
#! /usr/bin/python
import rospy
from std_msgs.msg import Float32MultiArray

def talker():
    pub_p = rospy.Publisher('mypoint', Float32MultiArray, queue_size=1)
    rospy.init_node('talker', anonymous=True)
    rate = rospy.Rate(10)
    while not rospy.is_shutdown():
	      array = [521,1314]
	      varpoint = Float32MultiArray(data=array)
        rospy.loginfo(varpoint)
	      pub_p.publish(varpoint)
        rate.sleep()
 
if __name__ == '__main__':
    try:
        talker()
    except rospy.ROSInterruptException:
        pass
